subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -48,6 +51,7 @@
 @Incubating
 public abstract class JacocoReportBase extends JacocoBase {
 
+    private Set<JacocoTaskExtension> taskExtensions = new HashSet<JacocoTaskExtension>();
```

Comment:
I don't think we'll want to merge the PR as is given the potential confusion for end users mentioned above. Regarding the other issues you mention can you please provide individual issues (and PR if you are interested?). We can discuss from there.